src/targets/gpu/device/include/migraphx/gpu/device/multi_index.hpp

```diff
@@ -31,7 +31,7 @@ struct multi_index
 };
 
 template <class ForStride>
-auto deduce_for_stride(ForStride fs) -> decltype(fs(id{}));
+__device__ __host__ auto deduce_for_stride(ForStride fs) -> decltype(fs(id{}));
```

Comment:
It seems strange to add `__device__ __host__` to this function when it is unevaluated. Maybe it would be better to use `constexpr` for this.